### PR TITLE
Improve net/buffer

### PR DESCRIPTION
**Pull request by bassosimone**
This pull request aims to simplify and to make more robust net/buffer code. More details in the individual commit messages. This is what was needed to prepare this part for version 0.1.

### CODE REVIEW

File: include/measurement_kit/net/buffer.hpp [hunk 1]
```diff
@@ -93,52 +47,48 @@ class Buffer : public NonCopyable, public NonMovable {
      */
 
     Buffer &operator<<(evbuffer *source) {
-        if (source == NULL)
-            throw std::runtime_error("source is NULL");
+        if (source == nullptr) throw std::runtime_error("source is nullptr");
         if (evbuffer_add_buffer(evbuf, source) != 0)
             throw std::runtime_error("evbuffer_add_buffer failed");
-        return (*this);
+        return *this;
```
Comment (hellais) [inline include/measurement_kit/net/buffer.hpp]: Just to understand, is this just a stylistic change or does it change the meaning?

Comment (hellais) [inline include/measurement_kit/net/buffer.hpp]: nevermind I didn't read the commit log (cosmetic changes)...

**Comment by hellais** [comment]
Skimming over the code I don't see any reason why not to merge. In the next days I will test it also by running it, but if I fail to do so soon, you should feel free to merge it.

**Comment by bassosimone** [comment]
Thank you!

**Comment by bassosimone** [comment]
Merging!

**Pull request merged**